### PR TITLE
add release docs

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
Why is this PR needed?

To have clear instructions on how to do the releases.

Replaces https://github.com/SUSE/skuba/pull/506

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -45,3 +45,19 @@ This list is not comprehensive, meaning if you want to include more detail than
...
+In order to create a new release:
+* Update the VERSION file:
+  echo "x.y.z" > VERSION
+* Commit with the message "Bump version: x.y.z", being x.y.z the version. Please use semantic versioning.
+  git commit -m "Bump version: x.y.z" -s
+* Tag with vx.y.z
+  git tag vx.y.z
+* Push
+  git push --tags origin master
+* Create the changelog:
```
Comment (stefsuse) [inline CONTRIBUTING.md]: we should add an extra tag for 'devel', 'alpha', and 'beta' so when we have a numbered only release it is implied to be a stable production release.

**Comment by jordimassaguerpla** [comment]
Test failures have nothing to do with this PR

**Comment by jordimassaguerpla** [comment]
@davidcassany @danielorf does this LGTY?

**Comment by jordimassaguerpla** [comment]
I added some changes we realized we need when doing the RC1 release.

**Comment by jordimassaguerpla** [comment]
rebasing. It was not passing the tests because the title was too long.

**Comment by danielorf** [comment]
Is it worth mentioning how to update kubernetes version (like [this commit](https://github.com/SUSE/skuba/commit/385a7a68c760c91e5f9d4da41ad357367a0c4436))?

This verges on internal documentation but updating k8s version is integral to many/most/all skuba releases.

**Comment by danielorf** [comment]
LGTM otherwise.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -46,3 +46,43 @@ This list is not comprehensive, meaning if you want to include more detail than
...
+    git commit -m "Bump version: x.y.z" -s
+
+Ask for a reviewer. Once reviewed, checkout master branch:
+
+    * git checkout master && git pull
+
+And tag:
+
+  * Tag with vx.y.z
+    git tag vx.y.z
```
Comment (ereslibre) [inline CONTRIBUTING.md]:
```suggestion
    git tag -s vx.y.z -m "Tag x.y.z"
```

I usually sign the tag as well, unsure if you want to force it that way.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -46,3 +46,43 @@ This list is not comprehensive, meaning if you want to include more detail than
...
+
+    * git checkout master && git pull
+
+And tag:
+
+  * Tag with vx.y.z
+    git tag -s vx.y.z -m "Tag x.y.z"
+
+* Push
+  git push --tags origin master
```
Comment (ereslibre) [inline CONTRIBUTING.md]: I usually do `git push upstream vx.y.z` instead; note that `origin` is usually the remote for the developer fork, whereas `upstream` is usually `github.com/suse/skuba`, but this depends a lot on every developer setup and preferences...

**Pull request merged**